src/models/HybridSystem.jl

```diff
@@ -0,0 +1,172 @@
+"""
+Representation of a Hybrid System that collects renewable generation, load, thermal generation
+and storage
+"""
+mutable struct HybridSystem{
+    T <: ThermalGen,
+    L <: ElectricLoad,
+    S <: Storage,
+    R <: RenewableGen,
+} <: StaticInjection
+    name::String
+    available::Bool
+    status::Bool
+    bus::Bus
+    active_power::Float64
+    reactive_power::Float64
+    thermal_unit::T
+    electric_load::L
+    storage::S
+    renewable_unit::R
+    # PCC Data
+    "Thermal limited MVA Power Output of the unit. <= Capacity"
+    pcc_impedance::ComplexF64
+    rating::Float64
+    input_active_power_limits::NamedTuple{(:min, :max), Tuple{Float64, Float64}}
+    output_active_power_limits::NamedTuple{(:min, :max), Tuple{Float64, Float64}}
+    reactive_power_limits::Union{Nothing, Min_Max}
+    base_power::Float64
+    operation_cost::OperationalCost
+    "corresponding dynamic injection device"
+    services::Vector{Service}
+    dynamic_injector::Union{Nothing, DynamicInjection}
+    ext::Dict{String, Any}
+    "internal forecast storage"
+    forecasts::InfrastructureSystems.Forecasts
+    "power system internal reference, do not modify"
+    internal::InfrastructureSystemsInternal
+end
```

Comment:
Per conversation with @yinongsun  and @bethanyfrew , we think that all of the PCC parameters and costs should be `Nothing`able. This allows the user to explicitly control the DC coupled vs AC coupled nature of the system along with the associated operational cost.

Reply:
ok, I will probably come back to this after merging 1.0